### PR TITLE
Fixes to enable soji to run on OS X

### PR DESCRIPTION
**Pull request by iacobus42**
I made a few changes to soji to enable it to run on OS X. They are 
- soji used the `--date` flag with `date`. This is a GNU extension to `date`. To enable it to work on OS X and other non-GNU *nix systems, the flags needed to be changed. `date --date` is easily replaced by `date -jf`. See the answer by Dirk Eddelbuettel and comments at http://stackoverflow.com/questions/1842634/parse-date-in-bash for more details
- OS X `date` requires the `-v` flag to add time to the present date
- `awk` in OS X would issue a warning when `-F ''` was used. To suppress this warning, I changed it to use `-FS=,`.
- OS X installs things in different locations. I changed the shebang at the start of `soji` and `soji.scm` to point to the guile at `/usr/local/bin` and not `/usr/bin`.(More on this later) OS X won't let you install to `/usr/bin` even with sudo unless you disable some OS protection features so it is often just easier to install to `/usr/local/bin` and deal with it.
- OS X won't let you install soji to `/usr/bin`. In the Makefile I changed the install location for soji to `/usr/local/bin/soji` from `/usr/bin/soji`
- When testing on OS X, cronjobs weren't properly running any code located in `/usr/local/bin` unless I provided the full path.
- `notify-send` doesn't work on OS X as far as I can tell. I used the terminal-notifier project by julienXX (https://github.com/julienXX/terminal-notifier) to provide for OS notifications based on the cron jobs in OS X
- I used the brightness project by nriley (https://github.com/nriley/brightness) to adjust the screen brightness using cron jobs. It only changes the brightness of the primary screen. On my MBP, only the laptop brightness changes. I have no idea how it works on a desktop. The external monitors do not change and as far as I can tell OS X has no provision for changing them.

I used the `uname` to determine if the system is on OS X or not and run the correct code accordingly (OS X if the `uname -a` contains `Darwin`). It only uses the OS X specific code when on OS X, otherwise it runs the original soji code. 

Since OS X and Ubuntu have different shebangs and paths required, I wrote a version of the files `soji`, `soji.scm` and `Makefile` with the appended OS (`for_linux` or `for_os_x`). I then wrote a very simple script to use `mv` the system specific files to the generic names (e.g., on OS X, `soji_for_os_x` becomes `soji`) and then run `make install`. A user should be able to install soji on either Ubuntu or OS X by just running `sudo sh install_soji`. 

I have tested the repo on Ubuntu and OS X El Capitan.

### CODE REVIEW

**Comment by joecannatti** [comment]
Thanks for all this work @iacobus42 !

I will review this PR in detail this evening, but it all looks good with a quick glance.

**Comment by iacobus42** [comment]
I added testing with travis to cover OS X but the OS X jobs don't seem to be starting on Travis. The OS X testing environment seems very (very) slow to start. Build 114 passed both - Build 115 (currently running) is the same as Build 114 but with an updated `README.md` covering the OS X-specific dependences.

File: subcommands/status [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+    fi
 fi
+
+
+
+
+
+
+
+#!/usr/bin/env bash
```
Comment (joecannatti) [inline subcommands/status]: Can you take out these blank lines and extra shebang?

File: subcommands/status [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
-if [ -e $(soji note-path log) ]; then
-  task=`cat $(soji note-path log) | egrep "^\*\* (start|break|lunch|meditation|meeting)" | tail -1 | sed 's/[^-]*-- //' | sed 's/\(.*\) --.*/\1/' |  awk -F '' '{if (NF > 20) {print substr($0, 0, 21)"..."} else print $0}'`
-  task=`echo $task | xargs`
-  start_time=`cat $(soji note-path log) | egrep "^\*\* (start|break|lunch|meditation|meeting)" | tail -1 | sed 's/.*-- //'`
-  if [ "$start_time" != "" ]; then
-    seconds=`expr $(date +'%s') - $(date +'%s' --date $start_time)`
-    minutes=`expr $seconds / 60`
-    echo "$task -- $minutes min"
-  fi
+if [ "$os_x" != "" ]; then
```
Comment (joecannatti) [inline subcommands/status:4]: Does this file need 

```
os_x=`uname -a | fgrep -i 'Darwin'` 
```

at the top, like the others?

File: subcommands/bright [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
-done
+os_x=`uname -a | fgrep -i 'Darwin'`
+
+#echo $os_x
 
+if [ "$os_x" != "" ]; then
+    /usr/local/bin/brightness -m 0.8
+else
+    for disp in `DISPLAY=:0 xrandr --verbose | grep '(normal' | cut -d' ' -f 1`; do
+    DISPLAY=:0 xrandr --output $disp --display ":0" --brightness 1 || true
```
Comment (joecannatti) [inline subcommands/bright]: indent that line under the `for` please

**Review by joecannatti — CHANGES_REQUESTED** [review]
Thanks so much! Just a couple of small comments, then I'll merge. I may take a pass at minimizing some of the difference between the OS X and Linux versions after the merge. But this is great.

File: subcommands/status (hunk 2 above)
Comment (iacobus42) [reply subcommands/status:4]: Yes, I actually discovered that this morning. My ~/.config/soji/subcommands had an outdated build on my machine from when I was just focused on getting the OS/X issues worked out. That is corrected in the current build.

File: subcommands/status (hunk 1 above)
Comment (iacobus42) [reply subcommands/status]: Removed.

File: subcommands/bright (hunk 3 above)
Comment (iacobus42) [reply subcommands/bright]: Done.

**Comment by joecannatti** [comment]
I've had trouble in the past with the OS X builds on Travis. I'm okay with merging it without that working and I can play around with it later.

**Comment by iacobus42** [comment]
I just pushed your requested edits. The OS X travis tests are passing but it takes about 15-20 minutes for the OS X jobs to even start.

Thanks!

**Review by joecannatti — APPROVED** [review]

**Pull request merged**